helm_chart/hookexecutor/sanity.go

```diff
@@ -0,0 +1,214 @@
+package hookexecutor
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+)
+
+type sanityExecutor struct {
+	*baseExcutor
+}
+
+func newSanityExecutor(
+	kubeClient kubernetes.Interface,
+) *sanityExecutor {
+	return &sanityExecutor{
+		baseExcutor: &baseExcutor{
+			kubeClient: kubeClient,
+		},
+	}
+}
+
+func (e *sanityExecutor) Execute() error {
+	logger.Info("Performing actions in sanity test")
+
+	var err error
+	err = e.createSanityResources()
+	if err != nil {
+		return logger.ErrorRet(err, "Failed performing actions in sanity test")
+	}
+
+	err = e.deleteSanityResources()
+	if err != nil {
+		return logger.ErrorRet(err, "Failed performing actions in sanity test")
+	}
+
+	logger.Info("Successfully performed actions in sanity test")
+	return nil
+}
+
+// createSanityResources creates a pvc and a pod and wait utill they reach the desired state.
+func (e *sanityExecutor) createSanityResources() error {
+	logger.Info("Creating sanity resources")
+
+	var err error
+	pvc, pod := getSanityPvcAndPod()
+
+	err = updateStorageClassInPvc(pvc)
+	if err != nil {
+		return logger.ErrorRet(err, "Failed to create sanity resources")
+	}
+
+	err = updateNamespace([]runtime.Object{pvc, pod})
+	if err != nil {
+		return logger.ErrorRet(err, "Failed to move the sanity resources to the specified namespace")
+	}
+
+	logger.Info("Creating sanity PVCs")
+	_, err = e.kubeClient.CoreV1().PersistentVolumeClaims(pvc.Namespace).Create(pvc)
+	if err != nil {
+		e.deletePvc(pvc)
+		return logger.ErrorRet(err, "Failed to create sanity pvcs")
+	}
+
+	pvcWatcher, err := generatePvcWatcher(pvc.Name, pvc.Namespace, e.kubeClient.CoreV1())
+	if err != nil {
+		e.deletePvc(pvc)
+		return logger.ErrorRet(err, "Failed generating PVC watcher")
+	}
+	_, err = Watch(pvcWatcher, func(obj runtime.Object) bool {
+		pvc := obj.(*corev1.PersistentVolumeClaim)
+		return pvc.Status.Phase == corev1.ClaimBound
+	})
+	if err != nil {
+		e.deletePvc(pvc)
+		return logger.ErrorRet(err, "Failed waiting for PVC to be bound")
+	}
+
+	logger.Info("Creating sanity Pods")
+	_, err = e.kubeClient.CoreV1().Pods(pod.Namespace).Create(pod)
+	if err != nil {
+		e.deletePod(pod)
```

Comment:
Olga suggested that its better to keep the objects if it fail to create them, and the fail error should tell the user to check the pvc\pv\pod  sanity status to troubleshooting and also clean them up manually before run sanity again.
please do so for all the other failes.